### PR TITLE
Add warning regarding validation_epoch_end, test_epoch_end and ddp

### LINKED ISSUE #7691: Add warning for "ddp" in validation_epoch_end
## 📚 Documentation

As I currently understand, in `ddp` `validation_epoch_end` only sees part of the data which is located on selected device. This needs to be made more apparent in the documentation, since currently it only talks about `dp` and `ddp2` which made me think that for everything else it is done automatically.

Thanks!

### PR DESCRIPTION
**Pull request by Rizhiy**
## What does this PR do?

Fixes #7691

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -881,6 +881,9 @@ def validation_epoch_end(self, outputs: EPOCH_OUTPUT) -> None:
         Return:
             None
 
+        .. warning:: If you are using `ddp*` accelerators, each process will only see its own data.
```
Comment (awaelchli) [inline pytorch_lightning/core/lightning.py:884]: Why a warning? There is nothing misbehaving that needs to be warned about. 
In DDP the data is always partitioned across the devices, that's why it's called data parallel. And it doesn't matter if training, validation or test. I would rather modify the args description, i.e., change

> outputs: List of outputs you defined in ...

to

> outputs: List of outputs gathered from :meth:`validation_step` in the current process, or ...

**Review by awaelchli — CHANGES_REQUESTED** [review]

File: pytorch_lightning/core/lightning.py (hunk 1 above)
Comment (Rizhiy) [reply pytorch_lightning/core/lightning.py]: Changed to Note

Comment (carmocca) [reply pytorch_lightning/core/lightning.py]: +1 to updating the args description instead. Note/warning blocks are really distracting when reading the docs IMO

**Comment by Rizhiy** [comment]
Not done yet

**Pull request closed without merging**